code/modules/unit_tests/create_and_destroy.dm

```diff
@@ -200,6 +200,9 @@ GLOBAL_VAR_INIT(running_create_and_destroy, FALSE)
 			TEST_FAIL("[item.name] failed to respect force deletion [item.no_respect_force] times out of a total del count of [item.qdels]")
 		if(item.no_hint)
 			TEST_FAIL("[item.name] failed to return a qdel hint [item.no_hint] times out of a total del count of [item.qdels]")
+		if(LAZYLEN(item.extra_details))
+			var/details = item.extra_details.Join("\n")
+			TEST_FAIL("[item.name] failed with extra info: \n[details]")
```

Comment:
should put these into the failures part of the if chain, since you're losing it in tests

Reply:
losing it in tests?
I think having it separate is potentially useful, it opens up niche uses of the system and the cost is minor in comparison